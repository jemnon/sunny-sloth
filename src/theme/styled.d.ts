import 'styled-components';
import {
  Breakpoints,
  Colors,
  ColorMode,
  FocusColors,
  Fonts,
  FontSizes,
  Header,
  HoverColors,
  MaxWidths,
  Spacing,
  ZIndex,
} from '../types/theme';

declare module 'styled-components' {
  export interface DefaultTheme {
    breakpoints: Breakpoints;
    colors: Colors;
    focusColors: FocusColors;
    fonts: Fonts;
    fontSizes: FontSizes;
    header: Header;
    hoverColors: HoverColors;
    maxWidths: MaxWidths;
    spacing: Spacing;
    transition: string;
    zIndex: ZIndex;
  }
}
