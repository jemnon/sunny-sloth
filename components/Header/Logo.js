import React from 'react';

const Logo = () => (
  <img alt="whisper-of-yum" src="/static/logo-horizontal.svg" />
);

export default Logo;
