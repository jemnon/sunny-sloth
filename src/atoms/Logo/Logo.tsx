import React, { FC } from 'react';
import styled from 'styled-components';
import { ButtonReset } from '../Button';
import { Logo as LogoIcon } from '../Icons';
import { ColorMode } from '../../types/theme';

interface LogoProps {
  colorTheme?: ColorMode;
  onClick?: () => void;
}

const LogoContainer = styled.button`
  ${ButtonReset};
  cursor: pointer;
`;

const Logo: FC<LogoProps> = ({ colorTheme, onClick }) => (
  <LogoContainer onClick={onClick}>
    <LogoIcon
      fontSize="187px"
      fill={colorTheme === 'dark' ? '#fff' : '#111'}
      height="1.5rem"
      style={{ transition: 'all 0.47s ease' }}
    />
  </LogoContainer>
);

export default Logo;
