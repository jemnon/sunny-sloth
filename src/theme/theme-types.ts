export interface Breakpoints {
  sm: string;
  md: string;
  lg: string;
  xlg: string;
}

export interface Fonts {
  lato: string;
  latoBold: string;
  latoItalic: string;
  noto: string;
  notoBold: string;
}

export interface Colors {
  cream: string;
  darkTeal: string;
  teal: string;
  lightBrown: string;
  nearBlack: string;
  nearWhite: string;
  darkOrange: string;
  orange: string;
  white: string;
  red: string;
  green: string;
}

export type ColorMode = 'dark' | 'light';

export interface MaxWidths {
  sm: string;
  md: string;
  lg: string;
}

export interface Spacing {
  sm1: string;
  sm2: string;
  sm3: string;
  sm4: string;
  md1: string;
  md2: string;
  md3: string;
  md4: string;
  lg1: string;
  lg2: string;
  lg3: string;
  lg4: string;
}

export interface ZIndex {
  z0: string;
  z1: string;
  z2: string;
  z3: string;
  z4: string;
  z5: string;
  z6: string;
  z999: string;
  z9999: string;
}
