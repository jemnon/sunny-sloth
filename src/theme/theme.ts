const theme = {
  breakpoints: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
    xlg: '1440px',
  },
  fonts: {
    latoBold:
      `"Lato-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato-Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    notoBold:
      `"Noto Serif TC-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
  },
  spacing: {
    sm1: '0.25rem' /* 4px */,
    sm2: '0.5rem' /*  8px */,
    sm3: '0.75rem' /* 12px */,
    sm4: '1rem' /* 16px */,
    md1: '1.25rem' /* 20px */,
    md2: '1.5rem' /* 24px */,
    md3: '1.75rem' /* 28px */,
    md4: '2rem' /* 32px */,
    lg1: '2.25rem' /* 36px */,
    lg2: '2.5rem' /* 40px */,
    lg3: '2.75rem' /* 44px */,
    lg4: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
