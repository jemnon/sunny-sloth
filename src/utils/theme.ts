import { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
  breakpoints: {
    /* TODO: remove once conversion to styled-breakpoints is done */
    desktop: 'screen and (min-width: 960px)',
    tablet: 'screen and (min-width: 480px) and (max-width: 959px)',
    mobile: 'screen and (max-width: 479px)',
    /* ^^ */
    sm: '480px',
    md: '960px',
    lg: '1440px',
    xlg: '1920px',
  },
  fonts: {
    latoBold:
      `"Lato Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1440px',
  },
  spacing: {
    s1: '0.25rem' /* 4px */,
    s2: '0.5rem' /*  8px */,
    s3: '0.75rem' /* 12px */,
    s4: '1rem' /* 16px */,
    s5: '1.25rem' /* 20px */,
    s6: '1.5rem' /* 24px */,
    s7: '1.75rem' /* 28px */,
    s8: '2rem' /* 32px */,
    s9: '2.25rem' /* 36px */,
    s10: '2.5rem' /* 40px */,
    s11: '2.75rem' /* 44px */,
    s12: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
