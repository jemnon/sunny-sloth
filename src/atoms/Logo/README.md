# Logo

### Description
Component description

```js
import Logo from '@fenderdigital/ui/FILL IN TYPELogo';
```

### Props
* `propName` - property name and description 
