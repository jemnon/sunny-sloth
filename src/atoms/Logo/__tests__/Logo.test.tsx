import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Logo from '../Logo';
import theme from '../../../theme';

describe('<Logo />', (): void => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Logo />
    </ThemeProvider>
  );
  it('exists', (): void => {
    expect(Logo).toBeDefined();
  });
  it('renders defaults correctly', (): void => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});