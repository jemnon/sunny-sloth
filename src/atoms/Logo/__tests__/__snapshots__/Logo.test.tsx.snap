// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> renders defaults correctly 1`] = `
<button
  className="sc-dJjZJu jSRFKl"
>
  <svg
    className="sc-furvIG kRwNSd"
    fill="#111"
    fontSize="187px"
    height="1.5rem"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "transition": "all 0.47s ease",
      }
    }
    viewBox="0 0 256.94 32"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256.06 14.8c-1.38-.77-2.67 1-1.09 2.07 1.93 1.35 2.73-1.15 1.09-2.07zM251.65 13.91c-.75-.06-6.78-.11-9.77 0s-3.39-1.15-4.08-1.95a2.66 2.66 0 00-2.64-.92 8.8 8.8 0 00-3.39 1.49c-1.21.92-1.27.46-1.61.12a14 14 0 00-1.78-2.07c-1.15-1-2 0-2.7.69s-1 .29-1.67-.46a1.58 1.58 0 00-2-.35 34.94 34.94 0 00-3.39 2.42c-1.14 1-1.09-.35-1.43-1.15s-1.3-1.09-2.07.06a54.14 54.14 0 01-4.77 5.22c-.75.64-.86.06-.8-.51a16.14 16.14 0 011.6-4.6c.95-1.92-1.75-1.89-3-.17a21 21 0 01-4.54 4.48c-1.78 1.26-1.15-1.61-.12-4.14s-1.95-3.07-3.44-1c-.52.72-6.21 4.16-8.16 5.65s-1.64-.52-1.87-3.5c-.11-1.5-2.61-2.07-3.3-.66-.47 1-4.22 3.56-5.63 4-1.18.38-.92-.32-.72-1.23a9.58 9.58 0 011.53-3.54c1.06-1.6-.44-2.5-2.84-1.57s-16 4.63-16.58 4.78-.92.54-1-1.18-.57-4.56-.46-5.21-.56-3.47-1.96-2.24-7.47 5.36-8.5 5.63-1.15-.15-1.57-.34-2.93-1.29-3.5 1.5c-.82-.65-1.66-1.07-2.72-.2-3.27 2.7-11.32 2.24-12.41 2.12s-1.72.35-1.49-2.47-.69-5.51-1.15-6-1.41-.89-1.9.63c-.69 2.18-6.33 7.86-7.12 8.44-1.55 1.15-3.75.23-2.36-1.26 3.74-4 .41-5.32-.69-3.62-3.33 5.17-8.9 6.21-9.7 6.44s-1.38.74-1.32-1.21-1.5-4.94-2.59-5.4-1.78-.34-2.24.63a28 28 0 00-1.38 5.35c.06.4-.52.91-.46-.06s.52-2.36 0-3.79-2.29-.87-3 0-9.46 3.61-10.32 3.82-1.21-.29-1.73-1.73-1.66-4-2.7-4.65-1.72-.75-2.64.63-10.23 6-11.37 6.44-2 .23-2.48-1.55a33 33 0 00-1.26-4.71c-.34-.52-2.18-1.38-2.18.46s-6.22 5.8-6.49 5.9c-.46-3.5-1.15-6.66-2-7.64s-1.33-.4-2 .35a26.65 26.65 0 01-3.1 3.16c1-4.2 3.79-9.71 1.55-12.3s-1.28 1.08-1.86 2.37c-1.72 3.85-12.42 11.24-18.67 10.23-2.47-.41-1 1.43-.86 1.89-.81-.17-1.49.06-1.49.52s-3.57 4.48-4.48 5.06-1.23.17-1.15-1.44c0-1.27 1-6.49 2.12-8.39s-.72-3-2.07-1.66c-1 1-9.7 9.13-18.84 10-2 .2-.4 2.64 1.21 2.35 6.6-1.18 14.93-7 15.62-7.46s.69-.18.58.45-2.07 7.53 1.32 8.74 5.8-4 6.09-4.2c.92.92 2 2.07 3.45 2.24s1.6-1.78.34-5.45c9.08-.24 14.51-6.82 15.62-8.05s.69.57.64.91-3.11 7.41-3.85 8.68c-.44.74.1 3.64 2.1 1.32s5.4-5.28 5.75-5.74 1.2-1.67 1.49.17a58.29 58.29 0 001.09 6c.23.35.81 1.09 2 .29s6.21-3.74 6.9-4.89c.51 1.84 2.18 4.37 2.87 4.66a3.36 3.36 0 003.3-.17c1.61-.87 10.69-5.41 11.32-6.15s1-.64 1.32.11 4 4.67-2.74 4.22c-2.43-.17.93 3.18 6.36 1.7 2.61-.71 11.72-3.62 12.35-4 .17 2.12-.86 14.21-1.01 15.12-.17 1.32 1.9 2.87 2.25.05s3.29-14.54 3.38-15.39c.23-2 .89-2.22 1.33-1.49 1.43 2.41.05 4 0 5.45s1 1.38 1.6 1.21 8.22-2.13 9.25-2.93 1.15-.57 1.5.06 1.89 2.24 4.76.57 6.21-5.28 7.13-7.58c.73-1.84.8.23.8.8s-1.32 5.81 2.18 6.09c11.06.91 13-.62 13.85-1.38a2.28 2.28 0 012.21-.24c0 1.16.16 2.24.23 2.8.16 1.15 1.65 1.22 2.69.38s2.14-2.18 2.22-3.22 1.11-1.37 1.49-1.57 5.59-2.83 6.32-3.71.84-.31.92.54.27 3.75.31 4.21.19.8-.73 1.15-4.44 1.45-4.9 1.64-1.31.58-.65 1.65 1 2.37 2.6 2.8a23.94 23.94 0 014.44 1.53c.5.34 1.92.23 1.69-1.57s-.31-2.8-.35-3.18-.38-1 1-1.3c1.11-.26 11.22-3.43 15.64-4.4a13.14 13.14 0 00-.27 5.11c.35 1.26 1.84 1.26 3.22 1s5.4-3.45 6.49-4.31a18.88 18.88 0 00.35 3 1.86 1.86 0 01-.58 2c-.46.51-5.68 4.54-6.2 5.11s-.4 1.49-.06 2.53a10.36 10.36 0 003.55 3.59c1.15.64 2.48.28 3.41-1.2 1-1.65 1.88-6 2.05-9.19.92-.92 6.89-5 8.62-6.27s1.2.81 1.32 1.79.28 2.52 1.49 2.64a6.72 6.72 0 004.08-1.5c1.09-.86 1.21-.86 1.78 1s2.64 1.44 3.62 1.09 1.95-2.18 3.16-3.79.92-.11 1 1.09a1.16 1.16 0 002.06.64 22.36 22.36 0 015.35-5c.34.8 0 2.3 0 3.68s2.3 1.49 2.13 0c-.15-1.26 1.61-2.76 2.93-3.74a7.79 7.79 0 011.49 2.53 1.32 1.32 0 002.07.57 18.56 18.56 0 013.22-2.12c1.72-.86 2.58 1 3.21 1.66s3.62 1.21 4.89 1.15 6.49-.23 7.41.23 1.55.06 2.53-.75-.15-1.91-.89-1.97zm-223.76 5a1.65 1.65 0 01-.75-1.78c.14-.87.5-.36.56-.21s.46 1.34.53 1.53.37.92-.34.45zm120.24-3.54c-.15.19-.46.77-.59 1-.35-.73-.47-1.48.26-1.54s.58.26.33.54zm7.87 4.69c-.12-.11-.21-.31.13-.4s2.84-1 3.43-1.11c.06.63.19 2.51.19 2.68a31.7 31.7 0 01-3.75-1.14zm30.44 9.31c-.8 1.27-2.79-.63-3.15-1.31a1.93 1.93 0 01.24-1.77 27 27 0 014.45-3.73c.57 2.75-.92 5.75-1.54 6.81z"
    />
    <path
      d="M57.58 5.72a1.08 1.08 0 002 .58c.63-.92-.29-1.67-.63-1.84s-1.6-.63-1.37 1.26z"
    />
  </svg>
</button>
`;
